controllers/ie_controller.go

```diff
@@ -0,0 +1,81 @@
+package controllers
+
+import (
+	"fmt"
+	"net/http"
+	"reflect"
+
+	mgo "gopkg.in/mgo.v2"
+
+	"github.com/gin-gonic/gin"
+)
+
+// IEController Wrapped for Patient Collection
+type IEController struct {
+	DB             *mgo.Database
+	CollectionName string
+	Item           interface{}
+	IDScope        string
+}
+
+// All List All Items
+func (ie *IEController) All(c *gin.Context) {
+	items := ie.itemSlice()
+	ie.getCollection().Find(nil).All(items)
+	c.JSON(http.StatusOK, items)
+}
+
+// Create a Patient resource
+func (ie *IEController) Create(c *gin.Context) {
+	err := ie.getCollection().Insert(getJSONBody(c))
+	handleMongoError(c, err)
+}
+
+// Read Find a Patient
+func (ie *IEController) Read(c *gin.Context) {
+	item := ie.item() //ie.Item
+
+	ie.getCollection().FindId(c.Param("id")).One(item)
+
+	if reflect.DeepEqual(item, ie.item()) {
```

Comment:
It doesn't return a nil, it returns an empty struct (since that is the reference that is passed in). This is why I had to use the deep equal.